### PR TITLE
Modify supertux-appdata.patch to fix broken metainfo validations

### LINKED ISSUE #20: Flathub lists Supertux as a proprietary app
The Flathub page incorrectly says this is a proprietary app.

### PR DESCRIPTION
**Pull request by Zishan-Rahman**
Hopefully fixes #20. Again.

### CODE REVIEW

File: org.supertuxproject.SuperTux.json [hunk 1]
```diff
@@ -96,6 +96,9 @@
                 "-DCMAKE_BUILD_TYPE=Release",
                 "-DIS_SUPERTUX_RELEASE=ON"
             ],
+            "post-install": [
+                "install updated-appdata.xml /app/share/appdata/${FLATPAK_ID}.appdata.xml"
```
Comment (Newbytee) [inline org.supertuxproject.SuperTux.json]:
```suggestion
                "install -Dm644 updated-appdata.xml /app/share/metainfo/${FLATPAK_ID}.appdata.xml"
```

Comment (Zishan-Rahman) [reply org.supertuxproject.SuperTux.json]: I somehow missed your suggested change when I actually made your changes, but I just made your changes:
https://github.com/flathub/org.supertuxproject.SuperTux/pull/23/commits/ed8a10e532bf24829c0f0201dc2f7a987b2bb008
https://github.com/flathub/org.supertuxproject.SuperTux/pull/23/commits/fc68baa86cf297e841fe6f495809d930705833f1

**Comment by Zishan-Rahman** [comment]
So it's saying it can't find a suitable 128x128 icon to use, but shouldn't it be able to use [the scalable SVG file](https://github.com/SuperTux/supertux/blob/v0.6.3/supertux2.svg) in its place? Usually that works without issue. I'll try building it again just to make sure.

**Comment by Zishan-Rahman** [comment]
bot, build

**Comment by Zishan-Rahman** [comment]
image

So [_this_ build](https://buildbot.flathub.org/#/builders/6/builds/82273) failed for exactly the same reason. I guess we'll have to convert it to a 128x128 PNG specifically for this Flatpak, then.

**Comment by Zishan-Rahman** [comment]
Yeah, I forgot to change the SVG filename when I took and modified the SVG conversion instructions from the Godot 4 Flatpak manifest 🤦‍♂️

I'm doing that now

File: org.supertuxproject.SuperTux.json [hunk 2]
```diff
@@ -96,6 +96,11 @@
                 "-DCMAKE_BUILD_TYPE=Release",
                 "-DIS_SUPERTUX_RELEASE=ON"
             ],
+            "post-install": [
+                "install -Dm644 updated-appdata.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml",
+                "rsvg-convert supertux2.svg -w 128 -h 128 -a -f png -o 128.png",
```
Comment (razzeee) [inline org.supertuxproject.SuperTux.json:101]: No idea, if this fixes your problem, but I would recommend to only copy the svg over as an icon and don't convert - as it's much more futureproof.

```
install -Dm644 supertux2.svg /app/share/icons/hicolor/scalable/apps/org.supertuxproject.SuperTux.svg
```

Should hopefully work

Comment (hfiguiere) [reply org.supertuxproject.SuperTux.json:101]: you don't need this. a SVG is enough

File: org.supertuxproject.SuperTux.json [hunk 3]
```diff
@@ -108,9 +113,11 @@
...
                 },
                 {
-                    "//": "Remember to delete this patch when the next SuperTux version gets released, as the appdata fix has also been merged into SuperTux's source code: https://github.com/SuperTux/supertux/pull/2668",
-                    "type": "patch",
-                    "path": "supertux-appdata.patch"
+
+                    "type": "file",
+                    "url": "https://raw.githubusercontent.com/SuperTux/supertux/12cf6c012e2bc1b91ef12fe56df1b169b2a5fa47/supertux2.appdata.xml",
+                    "sha256": "f6e87e02dd11037fdade5e65ebb8c956140129e71dd4d00ec9435fc1dab1f096",
+                    "dest-filename": "updated-appdata.xml"
```
Comment (hfiguiere) [inline org.supertuxproject.SuperTux.json:120]: this completely ignore line 9. which is why things aren't working the way want.

what's wrong with the patch?

File: org.supertuxproject.SuperTux.json (hunk 2 above)
Comment (razzeee) [reply org.supertuxproject.SuperTux.json:101]: But that needs to be in some folder?

File: org.supertuxproject.SuperTux.json (hunk 3 above)
Comment (Zishan-Rahman) [reply org.supertuxproject.SuperTux.json:120]: The updated appdata file linked here has some other changes made to it, besides the one made by the patch, that help fix [the metainfo validation errors described here](https://github.com/flathub/org.supertuxproject.SuperTux/issues/20#issuecomment-1816947323).

File: org.supertuxproject.SuperTux.json (hunk 2 above)
Comment (hfiguiere) [reply org.supertuxproject.SuperTux.json:100]: this should be removed, its unecessary.

File: org.supertuxproject.SuperTux.json (hunk 3 above)
Comment (hfiguiere) [reply org.supertuxproject.SuperTux.json:119]: then the patch should have been updated instead

Comment (Zishan-Rahman) [reply org.supertuxproject.SuperTux.json:119]: > then the patch should have been updated instead

Ah. Didn't even think of that :man_facepalming:

**Comment by Zishan-Rahman** [comment]
Per [Hubert's advice](https://github.com/flathub/org.supertuxproject.SuperTux/pull/23#discussion_r1398515525), I will modify the appdata patch instead of linking the new file to download. I will also rename this PR accordingly. I've kept the prior commit history (before the forthcoming force push) [in this branch](https://github.com/Zishan-Rahman/org.supertuxproject.SuperTux/tree/backup-1).

**Comment by Zishan-Rahman** [comment]
Looks like the patch was applied successfully. I checked just to make sure. Everything else, including the icons, should be exactly as they were before, so it looks like this PR can be merged :+1:

@hfiguiere This is irrelevant to this Flatpak specifically, but [for a Flatpak I'm trying to get to Flathub](https://github.com/flathub/flathub/pull/4595), I'd like to take the `xvfb-module` folder from [another Flatpak I recently contributed to](https://github.com/flathub/io.github.trevorsandy.LPub3D/) and use it there for a very important build step. I'm not exactly sure how Flatpak manifests on Flathub are licensed for this kind of usage (taking multiple files from other manifests like this), so I want to make definitively sure I'm able to use said module without getting into trouble somehow. I requested permission from you to do this, and/or clarification on whether I need permission to do this, a few times in the past few weeks. I'd like to kindly reiterate my request here, as it would save me some time and allow me to push some additional important changes I made to it as soon as possible.

**Review by Newbytee — APPROVED** [review]
Thanks a bunch!

**Pull request merged**